### DISCUSSION TITLE
(Tree)DataGrid Cell Selection

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#10850] [Q&A]
Is there any way to select single cells in a DataGrid? I also had a look at TreeDataGrid but it doesn't seem to support it either.
We have a requirement for our current project so I need to somehow make it work.

**Comment by timunie** [comment]
Hm @SD-alexander-bittermann I feel like this is not yet possible with DataGrid. If it will ever work, I cannot say as it is taken from Silverlight and not WPF. TreeDataGrid may have this in the future, but I don't know the current state of it. I think you could ask for an offer if you need such a feature urgently (if you have a budget at all for it)

**Reply by ghost** [reply]
Thanks for the reply. 
That's very unfortunate :(
Right now I'm trying to write my own SelectionModel, let's see if I succed with that. Otherwise we'll probably have to think about asking for an offer...

**Reply by timunie** [reply]
Probably also check: https://github.com/wieslawsoltes/DataBox

**Reply by ghost** [reply]
Thanks for the tip, I had a quick look but couldn't find anything that implies cell selection support. Could you maybe point me in the right direction?

**Reply by timunie** [reply]
In that case there is none. I never used this control on my own, just remembered that is exits and thought it's worth a look.

**Reply by ghost** [reply]
Thanks for the reply!
I managed to write a SelectionModel for TreeDataGrid that works for my requirements, so no problem for now.

**Reply by KrzysztofDusko** [reply]
@SD-alexander-bittermann can you share your sollution ?

**Reply by DevAM-Tools** [reply]
Maybe this could help for cell selection with TreeDataGrid:

https://github.com/AvaloniaUI/Avalonia/discussions/14262
